lib/beaker/host/unix/file.rb

```diff
@@ -29,4 +29,196 @@ def file_exist?(path)
     result = exec(Beaker::Command.new("test -e #{path}"), :acceptable_exit_codes => [0, 1])
     result.exit_code == 0
   end
+
+  # Gets the config dir location for package information
+  #
+  # @raise [ArgumentError] For an unknown platform
+  #
+  # @return [String] Path to package config dir
+  def package_config_dir
+    case self['platform']
+    when /fedora|el-|centos/
+      '/etc/yum.repos.d/'
+    when /debian|ubuntu|cumulus/
+      '/etc/apt/sources.list.d'
+    else
+      msg = "package config dir unknown for platform '#{self['platform']}'"
+      raise ArgumentError, msg
+    end
+  end
+
+  # Returns the repo filename for a given package & version for a platform
+  #
+  # @param [String] package_name Name of the package
+  # @param [String] build_version Version string of the package
+  #
+  # @raise [ArgumentError] For an unknown platform
+  #
+  # @return [String] Filename of the repo
+  def repo_filename(package_name, build_version)
+    variant, version, arch, codename = self['platform'].to_array
+    variant = 'el' if variant == 'centos'
+
+    repo_filename = "pl-%s-%s-" % [ package_name, build_version ]
+    case variant
+    when /fedora|el|centos/
+      fedora_prefix = ((variant == 'fedora') ? 'f' : '')
+      pattern = "%s-%s%s-%s.repo"
+      pattern = "repos-pe-#{pattern}" if self.is_pe?
+
+      repo_filename << pattern % [
+        variant,
+        fedora_prefix,
+        version,
+        arch
+      ]
+    when /debian|ubuntu|cumulus/
+      repo_filename << "%s.list" % [ codename ]
+    else
+      msg = "#repo_filename: repo filename pattern not known for platform '#{self['platform']}'"
+      raise ArgumentError, msg
+    end
+
+    repo_filename
+  end
+
+  # Gets the repo type for the given platform
+  #
+  # @raise [ArgumentError] For an unknown platform
+  #
+  # @return [String] Type of repo (rpm|deb)
+  def repo_type
+    case self['platform']
+    when /fedora|el-|centos/
+      'rpm'
+    when /debian|ubuntu|cumulus/
+      'deb'
+    else
+      msg = "#repo_type: repo type not known for platform '#{self['platform']}'"
+      raise ArgumentError, msg
+    end
+  end
+
+  # Tests for a package repo at the platform's path, returning it if it exists
+  #
+  # @param [String] buildserver_url URL for the buildserver
+  # @param [String] package_name Name of the package
+  # @param [String] build_version Version of the package
+  # @param [String] repo_name Name of the repo to check for the package in
+  #
+  # @return [String, nil] Path to the package repo if it exists, nil otherwise
+  def repo_path_exists?(buildserver_url, package_name, build_version, repo_name)
```

Comment:
The `?` suggests this returns a boolean, but instead it returns `[String, nil]`. This should be renamed, something like `get_repo_path`.